GestorDeEstoque/Controllers/ProdutoController.cs

```diff
@@ -2,64 +2,92 @@
         {
+            _context = context;
             _produtoRepository = produtoRepository ?? throw new ArgumentNullException(nameof(produtoRepository));
-
+            _produtoEstoqueRepository = produtoEstoqueRepository ?? throw new ArgumentNullException(nameof(produtoEstoqueRepository));
         }
 
-        [HttpGet("{id}")]
-        public async Task<IActionResult> BuscarProdutoPorId(int id)
+        [HttpGet("{produtoId}/estoqueId/{estoqueId}")]
+        public async Task<IActionResult> BuscarProdutoPorId(int produtoId, int estoqueId)
         {
-            var produto = await _produtoRepository.BuscarProdutoPorIdAsync(id);
-            if (produto == null)
+            try
             {
-                return NotFound();
+                var produto = await _produtoRepository.BuscaPorIdProdutoEhIdEstoqueAsync(produtoId, estoqueId);
+                if (produto == null)
+                {
+                    return NotFound(new { mensagem = "Produto ou estoque não encontrado" });
+                }
+                return Ok(produto);
             }
-            else
+            catch (Exception ex)
             {
-                return Ok(produto);
+                return BadRequest(new { mensagem = ex.Message });
             }
         }
 
 
         [HttpPost]
         public async Task<IActionResult> InserirProduto([FromBody] ProdutoDTO novoProdutoDTO)
         {
-            if (!ModelState.IsValid)
-            {
-                return BadRequest(ModelState);
-            }
-            var produto = new Produto
+            using var transaction = _context.Database.BeginTransaction();
+            try
             {
+                if (!ModelState.IsValid)
+                {
+                    return BadRequest(ModelState);
+                }
+                var produto = new Produto
+                {
+
+                    Nome = novoProdutoDTO.Nome,
+                    Descricao = novoProdutoDTO.Descricao,
+                    Preco = novoProdutoDTO.Preco
+                };
 
-                Nome = novoProdutoDTO.Nome,
-                Descricao = novoProdutoDTO.Descricao,
-                Preco = novoProdutoDTO.Preco,
-                Quantidade = novoProdutoDTO.Quantidade,
-                EstoqueId = novoProdutoDTO.EstoqueId
-            };
-            await _produtoRepository.InserirProdutoAsync(produto);
+                await _produtoRepository.InserirProdutoAsync(produto);
+                await _context.SaveChangesAsync();
 
-            return CreatedAtAction(nameof(BuscarProdutoPorId), new { id = produto.Id }, produto);
+                var produtoEstoque = new ProdutoEstoque
+                {
+                    ProdutoId = produto.Id,
+                    EstoqueId = novoProdutoDTO.EstoqueId,
+                    Quantidade = novoProdutoDTO.Quantidade
+                };
+
+                await _produtoEstoqueRepository.AdicionarQuantidadeAsync(produtoEstoque);
```

Comment:
Esse método apenas adiciona quantidade ou insere um novo produto estoque. 
Talvez seja melhor alterar o nome se for o caso.